dashboard/src/modules/components/SidebarComponent/index.jsx

```diff
@@ -26,54 +42,38 @@ const Menu = () => {
 
   useEffect(() => {
     if (pathname) {
-      const modPathName = pathname.replace(/\/+$/, "");
-      const currPath = modPathName.split("/").at(-1);
+      const currPath = pathname.replace(/^.*[/]([^/]+)[/]*$/, "$1");
 
       dispatch(setActiveItem(currPath));
     }
-  }, [dispatch, isLoggedIn, pathname]);
+  }, [dispatch, pathname]);
+
   return (
-    <>
+    <Nav onSelect={onSelect}>
       {isLoggedIn ? (
-        <Nav onSelect={onSelect}>
-          {menuOptions.map((item, index) => (
-            <NavGroup
-              key={index}
-              aria-label={item.group.title}
-              title={item.group.title}
-            >
-              {item.submenu.map((option) => (
-                <NavItem
-                  preventDefault
-                  onClick={() => navigate(option.link)}
-                  key={option.key}
-                  itemId={option.key}
-                  isActive={activeItem === option.key}
-                >
-                  {option.name}
-                </NavItem>
-              ))}
-            </NavGroup>
-          ))}
-        </Nav>
+        menuOptions.map((item, index) => (
+          <NavGroup
+            key={index}
+            aria-label={item.group.title}
+            title={item.group.title}
+          >
+            <MenuItem
+              data={item.submenu}
+              context={navigate}
+              activeItem={activeItem}
+            />
```

Comment:
You're not using the `context` prop...do you still want to pass it?

(What is the difference between what `useNavigate()` returns in the caller and what `useOutletContext()` returns in the callee?)

Reply:
The useNavigate hook returns a function that lets us to navigate the desired page.

The `Outlet`  provides a React context that can be accessed via the [useOutletContext](https://reactrouter.com/docs/en/v6/api#useoutletcontext) hook since it's a common enough pattern for parent routes to manage some state or value to be shared with children routes. 

But here, as I'm not using `Outlet` so it's no point in passing the context.